### PR TITLE
common: add coding style document

### PR DESCRIPTION
**Pull request by arnout**
We can update this document as we decide on the coding style.

Also refer to this new document from CONTRIBUTING.md.

### CODE REVIEW

File: CODINGSTYLE.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+### Naming convention
+
+* Almost everything uses `lower_snake_case`.
+* `#define` uses `UPPER_SNAKE_CASE`.
+* Everything is in a namespace.
+* Therefore, there are no prefixes.
+* Private data members have an "m" prefix: `m_data`.
+* Getter functions use the name without "m": `data()`.
+* Setter functions use the "set" prefix: `set_data(value)`.
```
Comment (tomereli) [inline CODINGSTYLE.md:23]: what about classes and structs? They are using CamelNotation, we should add this also

File: CODINGSTYLE.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# prplMesh coding style
+
+This document describes the intended coding style.
+Note that existing code does not always comply to it.
+
+Line length is 100 characters.
```
Comment (tomereli) [inline CODINGSTYLE.md]: We should allow some exceptions - for example tables and anything that looks hideous in 100 column boundary can skip clang-format by adding `clang-format-off` and `clang-format-on` surrounding it.

File: CODINGSTYLE.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+# prplMesh coding style
+
+This document describes the intended coding style.
+Note that existing code does not always comply to it.
```
Comment (tomereli) [inline CODINGSTYLE.md]: >", but is in the process of aligning to it - this is work in progress."

**Review by tomereli — CHANGES_REQUESTED** [review]

File: CODINGSTYLE.md (hunk 1 above)
Comment (arnout) [reply CODINGSTYLE.md:23]: If you exclude the generated code, most classes use lower snake case.

**Review by tomereli — APPROVED** [review]

**Pull request merged**